core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -7469,7 +7469,14 @@ RelExpr *Scan::bindNode(BindWA *bindWA)
      bindWA->setErrStatus();
      return NULL;
   }
-
+  if (naTable->hasLobColumn() && isSampleScan())
```

Comment:
Update statistics right now returns an error when a table has lob columns. Will open a JIRA for that. But your question on this also showed that this if check is a bit too restrictive. Will submit a change to move the check from binder phase to the generator phase where it's easier to detect sample columns alone.